tests/tools/test_system_info.py

```diff
@@ -1,72 +1,604 @@
+            """Mock execute_command to return fixture data."""
+            if command == ["free", "-b"]:
+                return (0, mock_free_output, "")
+            return (1, "", "Command not found")
+
+        # Expected values from mock data
+        # Percent is calculated as (used / total * 100)
+        # For RAM: 2513342464 / 6183096320 * 100 = 40.64860603691841
+        expected_memory_stats_json = {
+            "ram": {
+                "total": 6183096320,
+                "used": 2513342464,
+                "free": 1560502272,
+                "available": 3669753856,
+                "buffers": 2362519552,
+                "percent": 2513342464 / 6183096320 * 100,  # 40.64860603691841
+            },
+            "swap": {
+                "total": 6182400000,
+                "used": 0,
+                "free": 6182400000,
+                "percent": 0.0,
+            },
+        }
+
+        # Patch execute_command in processing module (used by default Rummager._collect)
+        with patch(
+            "linux_mcp_server.processing.execute_command",
+            new=AsyncMock(side_effect=mock_execute_command),
+        ):
+            result = await mcp.call_tool("get_memory_information", arguments={})
+
+            # MCP call_tool returns a tuple of (content, structured_output)
+            assert isinstance(result, tuple)
+            assert len(result) == 2
+            content, structured_output = result
+
+            # Content should be a TextContent object array
+            assert isinstance(content, list)
+            assert all(isinstance(item, TextContent) for item in content)
+
+            # Extract text from content and verify JSON contains exact values
+            assert isinstance(content[0], TextContent)
+            text = content[0].text
+            data = json.loads(text)
+
+            # Verify RAM information in JSON
+            assert "ram" in data
+            ram = data["ram"]
+            assert ram is not None
+            assert ram["total"] == expected_memory_stats_json["ram"]["total"]
+            assert ram["used"] == expected_memory_stats_json["ram"]["used"]
+            assert ram["free"] == expected_memory_stats_json["ram"]["free"]
+            assert ram["available"] == expected_memory_stats_json["ram"]["available"]
+            assert ram["buffers"] == expected_memory_stats_json["ram"]["buffers"]
+            assert ram["percent"] == pytest.approx(expected_memory_stats_json["ram"]["percent"], rel=1e-2)
+
+            # Verify swap information in JSON
+            assert "swap" in data
+            swap = data["swap"]
+            assert swap is not None
+            assert swap["total"] == expected_memory_stats_json["swap"]["total"]
+            assert swap["used"] == expected_memory_stats_json["swap"]["used"]
+            assert swap["free"] == expected_memory_stats_json["swap"]["free"]
+            assert swap["percent"] == expected_memory_stats_json["swap"]["percent"]
```

Comment:
```suggestion
            assert "ram" in data
            ram = data["ram"]
            assert ram["total"] == expected_memory_stats_json["ram"]["total"]
            assert ram["used"] == expected_memory_stats_json["ram"]["used"]
            assert ram["free"] == expected_memory_stats_json["ram"]["free"]
            assert ram["available"] == expected_memory_stats_json["ram"]["available"]
            assert ram["buffers"] == expected_memory_stats_json["ram"]["buffers"]
            assert ram["percent"] == pytest.approx(expected_memory_stats_json["ram"]["percent"], rel=1e-2)

            # Verify swap information in JSON
            assert "swap" in data
            swap = data["swap"]
            assert swap["total"] == expected_memory_stats_json["swap"]["total"]
            assert swap["used"] == expected_memory_stats_json["swap"]["used"]
            assert swap["free"] == expected_memory_stats_json["swap"]["free"]
            assert swap["percent"] == expected_memory_stats_json["swap"]["percent"]
```

Same here. If "ram" or "swap" is not present in `data`, the data["ram"/"swap"] key access would fail